working-w-copilot/copilot-in-projects-refactoring.md

```diff
@@ -115,240 +145,165 @@ Remember, the tests generated during the Copilot run in this lesson may not matc
-          response = client.post("/books", json=test_data)
-
-  def test_create_one_book_with_extra_keys(client, two_saved_books):
-      # Arrange
-      test_data = {
-          "extra": "some stuff",
-          "title": "New Book",
-          "description": "The Best!",
-          "another": "last value"
-      }
-
-      # Act
-      response = client.post("/books", json=test_data)
-      response_body = response.get_json()
-
-      # Assert
-      assert response.status_code == 201
-      assert response_body == "Book New Book successfully created"
-
-  def test_validate_book_with_valid_id(two_saved_books):
-      # Arrange
-      book_id = 1
-
-      # Act
-      result = validate_book(book_id)
-
-      # Assert
-      assert result.id == book_id
-
-  def test_validate_book_with_invalid_id():
-      # Arrange
-      book_id = "cat"
-
-      # Act & Assert
-      with pytest.raises(Exception) as e:
-          validate_book(book_id)
-          assert str(e.value) == "book cat invalid"
-
-  def test_validate_book_with_nonexistent_id():
-      # Arrange
-      book_id = 3
-
-      # Act & Assert
-      with pytest.raises(Exception) as e:
-          validate_book(book_id)
-          assert str(e.value) == "book 3 not found"
+
+  ...
+
+  def test_validate_book_succeeds(two_saved_books):
+    # Act
+    book = validate_book(1)
+
+    # Assert
+    assert book.id == 1
+    assert book.title == "Ocean Book"
+    assert book.description == "watr 4evr"
+
+  def test_validate_book_missing_record(two_saved_books):
+    # Act & Assert
+    with pytest.raises(HTTPException) as error:
+        validate_book(3)
+
+    response = error.value.response
+    assert response.status == "404 NOT FOUND"
+
+  def test_validate_book_invalid_id(two_saved_books):
+    # Act & Assert
+    with pytest.raises(HTTPException) as error:
+        validate_book("cat")
+
+    response = error.value.response
+    assert response.status == "400 BAD REQUEST"
   ```
 </details>
 
 #### Testing `update_book`
 
-With `validate_book` we were lucky, and Copilot outlined all of the scenarios that we intended to test, but that will not always be the case. We won't go step by step again, but let's apply the same approach we took with `validate_model` to testing our remaining dependencies `update_book` and `delete_book`.
+With `validate_book` we were lucky, and Copilot outlined all of the scenarios that we intended to test, but that will not always be the case. We won't go step by step for each dependency in this lesson, but let's apply the same approach we took with `validate_model` to testing our remaining dependencies `update_book` and `delete_book`.
 
-We'll start by using Copilot inline chat to ask for help generating tests for the `update_book` function. Here's what Copilot suggested:
+We'll start by using Copilot inline chat and the `/tests` shortcut to ask for help generating tests for the `update_book` function. Here's what Copilot suggested:
```

Comment:
Can we refer to `update_book` as an endpoint? Calling it just a function here souds like we're talking about another helper, like validate_model was.
```suggestion
We'll start by using Copilot inline chat and the `/tests` shortcut to ask for help generating tests for the `update_book` endpoint. Here's what Copilot suggested:
```